examples/two-tier/main.tf

```diff
@@ -49,27 +49,57 @@ resource "google_compute_instance" "www" {
     }
   }
 
-  metadata {
-    ssh-keys = "root:${file("${var.public_key_path}")}"
+  metadata = {
+    ssh-keys = "root:${file(var.public_key_path)}"
   }
 
   provisioner "file" {
-    source      = "${var.install_script_src_path}"
-    destination = "${var.install_script_dest_path}"
+    source      = var.install_script_src_path
+    destination = var.install_script_dest_path
 
     connection {
+      host = coalesce( # Simplify this to reference a specific desired IP address, if possible.
```

Comment:
This was a "TO-DO" that `terraform 0.12upgrade` generated.  My assumption is that it will stay this way as log as the access_config nat_ip is ephemeral, but please correct me if I'm wrong.

Reply:
I don't quite understand why we can't just grab the first access config.  It seems like we know for sure we're going to have one in this example.  Does this get executed at a counter-intuitive time?

Reply:
That could be - something like `self.network_interface.0.access_config.0.nat_ip`?

Reply:
That's my proposal, but I haven't tested it - I can't guarantee it'll work right.

Reply:
I've updated it, ran the plan and it worked.  Thanks for the review!